engine.go

```diff
@@ -160,3 +165,22 @@ func (e *engine) reset() {
 	e.previousActiveSegment = nil
 	e.activeBlock = nil
 }
+
+func trackTiming(showDebug bool, s *Segment) {
```

Comment:
I would rename this and make it part of `Segment`:

```golang
func (s *Segment) setStringValue(env environmentInfo, cwd string, ignoreFolders []string, debug bool) {
    err := s.mapSegmentWithWriter(env)
    if err != nil || s.hasValue(IgnoreFolders, cwd) {
	return
    }
    if debug {
        start := time.Now()
	defer (func() {
		s.active = true
		elapsed := time.Since(start)
		s.timing = elapsed
	})()
    }
    if s.enabled() {
	s.stringValue = s.string()
    }
}
```

We can then rewrite our for loop above to:

```golang
for _, segment := range segments {
	go func(s *Segment) {
		defer wg.Done()
                s.setStringValue(e.env, cwd, IgnoreFolders, showDebug)
	}(segment)
}
```